App/src/pages/CourseDetails.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useEffect, useState } from "react";
+import { useLocation } from "react-router-dom";
+import { Card, Heading, Text } from "@chakra-ui/react";
+
+function useQuery() {
+    return new URLSearchParams(useLocation().search);
+}
+
+const CourseDetails = () => {
+    const query = useQuery(); // Use the useQuery function
+
+    // Access specific query parameters
+    const subject: string = query.get("subject");
+    const catalogNumber: string = query.get("catalog");
+    const [data, setData] = useState([]);
+
+    useEffect(() => {
+        fetch("/data.json")
```

Comment:
I don't quite understand what you're fetching here. Is this a case of not having an endpoint available to use? If that's the case, you can make a relevant endpoint as part of the task or as a separate task.